klog_test.go

```diff
@@ -1365,6 +1365,74 @@ func TestErrorSWithLogr(t *testing.T) {
 	}
 }
 
+func TestCallDepthLogr(t *testing.T) {
+	logger := &callDepthTestLogr{
+		callDepth: -1,
+	}
+
+	testData := []struct {
+		msg           string
+		logFn         func()
+		expectedDepth int
+	}{
+		{
+			msg:           "Info log",
+			logFn:         func() { Info("info log") },
+			expectedDepth: 4,
+		},
+		{
+			msg:           "InfoDepth log",
+			logFn:         func() { InfoDepth(5, "infodepth log") },
+			expectedDepth: 8,
+		},
+		{
+			msg:           "InfoS log",
+			logFn:         func() { InfoS("infoS log") },
+			expectedDepth: 1,
+		},
+		{
+			msg:           "Warning log",
+			logFn:         func() { Warning("warning log") },
+			expectedDepth: 4,
+		},
+		{
+			msg:           "WarningDepth log",
+			logFn:         func() { WarningDepth(3, "warningdepth log") },
+			expectedDepth: 6,
+		},
+		{
+			msg:           "Error log",
+			logFn:         func() { Error("error log") },
+			expectedDepth: 4,
+		},
+		{
+			msg:           "ErrorDepth log",
+			logFn:         func() { ErrorDepth(7, "errordepth log") },
+			expectedDepth: 10,
+		},
+		{
+			msg:           "ErrorS log",
+			logFn:         func() { ErrorS(errors.New("some-error"), "errorS log") },
+			expectedDepth: 1,
+		},
+	}
+
+	for _, data := range testData {
+		t.Run(data.msg, func(t *testing.T) {
+			SetLogger(logger)
+			defer SetLogger(nil)
+			defer logger.reset()
+			defer logger.resetCallDepth()
+
+			data.logFn()
+
+			if data.expectedDepth != logger.callDepth {
```

Comment:
Testing callDepth this way is not very robust as it only test if values passed to CallDepth are not changed, but it doesn't validate the expected result that should come from using it. CallDepth should allow logger to find the stack frame that was responsible for calling the klog function. 

So if new intermediate function call is added in klog code, this code will not protect us from forgetting to update expected Depth.

Reply:
Changed around to asserting on caller location instead

I originally did not do this as it is still pretty fiddly, especially with the `*S` and `*Depth` functions